sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoComplete.java

```diff
@@ -0,0 +1,130 @@
+     *
+     * @param coreSubscriber The subscriber interested in the published items.
+     */
+    @Override
+    public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
+        Objects.requireNonNull(coreSubscriber, "'coreSubscriber' cannot be null.");
+
+        final AutoCompleteSubscriber<T> subscriber =
+            new AutoCompleteSubscriber<T>(coreSubscriber, onComplete, onAbandon, logger);
+
+        source.subscribe(subscriber);
+    }
+
+    static final class AutoCompleteSubscriber<T extends ServiceBusReceivedMessageContext> extends BaseSubscriber<T> {
+        private final CoreSubscriber<? super T> downstream;
+        private final Function<ServiceBusReceivedMessageContext, Mono<Void>> onComplete;
+        private final Function<ServiceBusReceivedMessageContext, Mono<Void>> onAbandon;
+        private final ClientLogger logger;
+
+        AutoCompleteSubscriber(CoreSubscriber<? super T> downstream,
+            Function<ServiceBusReceivedMessageContext, Mono<Void>> onComplete,
+            Function<ServiceBusReceivedMessageContext, Mono<Void>> onAbandon, ClientLogger logger) {
+            this.downstream = downstream;
+            this.onComplete = onComplete;
+            this.onAbandon = onAbandon;
+            this.logger = logger;
+        }
+
+        @Override
+        protected void hookOnSubscribe(Subscription subscription) {
+            logger.info("Subscription received. Subscribing downstream. {}", subscription);
+            downstream.onSubscribe(this);
+            requestUnbounded();
+        }
+
+        @Override
+        protected void hookOnNext(T value) {
+            final ServiceBusReceivedMessage message = value.getMessage();
+            final String sequenceNumber = message != null ? String.valueOf(message.getSequenceNumber()) : "n/a";
+
+            logger.verbose("Passing message downstream. sequenceNumber[{}]", sequenceNumber);
+
+            try {
+                downstream.onNext(value);
+                applyWithCatch(onComplete, value, "complete");
+            } catch (Exception e) {
+                logger.error("Error occurred processing message. Abandoning. sequenceNumber[{}]",
+                    sequenceNumber, e);
+
+                applyWithCatch(onAbandon, value, "abandon");
+            }
+        }
+
+        /**
+         * On an error, will pass the exception downstream.
+         *
+         * @param throwable Error to pass downstream.
+         */
+        @Override
+        protected void hookOnError(Throwable throwable) {
+            logger.error("Error occurred. Passing downstream.", throwable);
+            downstream.onError(throwable);
+        }
+
+        /**
+         * On the completion. Will pass the complete signal downstream.
+         */
+        @Override
+        protected void hookOnComplete() {
+            logger.info("Completed. Passing downstream.");
```

Comment:
verbose ? since it does not help in regular scenario.

Reply:
Sure